src/scheduler/batch/main.ts

```diff
@@ -61,18 +68,54 @@ export async function populateCurrentAnnualReview(lists: List[]): Promise<void>
   }
 }
 
+async function addAnnualReviewDateToPublishedLists(listsWithoutCurrentAnnualReviewDate: List[]) {
+  try {
+    const listsWithPublishedListItem = await Promise.all(
+      listsWithoutCurrentAnnualReviewDate.map(async (list) => {
+        const publishEventResult = await findFirstPublishedDateForList(list.id);
```

Comment:
Are you referring to combining the `findFirstPublishedDateForList` and `findListsWithoutNextAnnualReview`? Maybe I'm thinking ahead 🤔

I don't know if that's possible since there isn't a direct way to reference the Events table from the Lists one, unless there's a way to go through ListItems

Reply:
With prisma, you can
1. Query by "nested" or related 
2. Select/include by "nested" or related 
3. or some combination of the above

Here is how to do a nested query - if you just want to return the lists that are pass the criteria (it will just return the list - not the nested list items or events).

Some examples:

```.ts
 prisma.list.findMany({
    where: {
      items: {
        // Find the list where some items
        some: {
          history: {
            some: {
              // have an event
              type: "PUBLISHED", // of type published
              time: {
                // with the time of..
                //
              },
            },
          },
        },
      },
    },
  });
// Return the lists (without the related fields), where there are some items, that have an event of type published and time of.. 
```

```.ts
prisma.list.findMany({
    where: {
      nextAnnualReviewStartDate: {
        // find the list item with nextAnnualReviewStartDate
        //
      },
    },
    select: {
      // return
      items: {
        // the items
        where: {
          // where
          history: {
            some: {
              // there are some events
              type: "PUBLISHED", // with the type published.
              time: {
                // with the time of ..
              },
            },
          },
        },
      },
    },
  });

// find the list with an annualReviewStartDate of.. and return the related items where the history has some events that have the type published and a time of.. 
// (return value includes list and filtered list.items, but not list.items[].history). If you want to return history, you need to also select it. `select.items.select.history = true` (or nested query).
```


You can also do the reverse (starting with prisma.findMany.event, then select the related listItem, then list), but - since there are meant to be a lot of `Event` records, it's less performant to search this way. 

Generally, you want to narrow down your search as much as possible first (in the highest level `where` clause).

Have a look at `ListFindManyArgs` in `node_modules/.prisma/client/index.d.ts` (you should be able to just go to declaration). Usually I will navigate these types to figure out what's "available" to me to search against. 


ref: https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#querying-an-explicit-many-to-many